sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryRefreshTokenCache.cs

```diff
@@ -220,11 +220,13 @@ private async ValueTask GetRefreshTokenFromCredentialInBackgroundAsync(TaskCompl
 
         private async ValueTask<RefreshTokenInfo> GetRefreshTokenFromCredentialAsync(TokenRequestContext context, string service, bool async, CancellationToken cancellationToken)
         {
-            AccessToken aadAccessToken = async ? await _aadTokenCredential.GetTokenAsync(context, cancellationToken).ConfigureAwait(false) :
+            AccessToken aadAccessToken = async ?
+                await _aadTokenCredential.GetTokenAsync(context, cancellationToken).ConfigureAwait(false) :
                 _aadTokenCredential.GetToken(context, cancellationToken);
 
-            AcrRefreshToken acrRefreshToken = async ? await _authenticationRestClient.ExchangeAadAccessTokenForAcrRefreshTokenAsync(service, aadAccessToken.Token, cancellationToken).ConfigureAwait(false) :
-                _authenticationRestClient.ExchangeAadAccessTokenForAcrRefreshToken(service, aadAccessToken.Token, cancellationToken);
+            AcrRefreshToken acrRefreshToken = async ?
+                await _authenticationRestClient.ExchangeAadAccessTokenForAcrRefreshTokenAsync(PostContentSchemaGrantType.RefreshToken, service, null, null, aadAccessToken.Token, cancellationToken).ConfigureAwait(false) :
+                _authenticationRestClient.ExchangeAadAccessTokenForAcrRefreshToken(PostContentSchemaGrantType.RefreshToken, service, null, null, aadAccessToken.Token, cancellationToken);
```

Comment:
What should we pass for `tenant` and `refreshToken` in the service call?

Reply:
Discussed offline - the swagger has changed, as shown in this diff: https://www.diffchecker.com/HqF59yjW

The input refresh token should be the AAD refresh token (not the ACR refresh token), and I believe tenant should be the tenant of the authorized service principal.  If it's not clear where those are in the Azure.Identity APIs, @schaabs and/or @christothes should be able to help you out.

Reply:
tenantId can be null since the service will be able to get the value directly from the token.  GrantType should be accessToken. Refresh_token value can be null.

Reply:
Hadn't we decided before that it was better to use the AAD refresh token than the AAD access token, since it has a shorter lifetime?

Reply:
This call is basically exchanging the AAD access token that you got (via tokenCredential) to the refresh token which we will then cache.  Looking at the swagger - they seem to have unified the 2 calls. We earlier called an endpoint where grant_type was defaulted to accessToken and we did not have any parameters for refresh_token. We now call a unified endpoint that basically does the same (from what I can tell).

Reply:
> This call is basically exchanging the AAD access token that you got (via tokenCredential) to the refresh token which we will then cache. 

Yes, it is exchanging the AAD access token for the **ACR** refresh token.  When we last talked to @schaabs, I believe he had said we would rather not send the AAD access token around, and instead send the **AAD** refresh token.  Does this new service endpoint allow us to exchange the AAD refresh token for the ACR refresh token? If so, is there a way to send the AAD refresh token instead of the AAD access token?

Reply:
Scott has confirmed that I am incorrect :)

Reply:
Thanks, Anne, for ensuring the behavior is correct. And thanks to Pallavi and Scott for their answers.